src/mscorlib/src/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -575,10 +446,10 @@ public void SetResult(TResult result)
         {
             // Get the currently stored task, which will be non-null if get_Task has already been accessed.
```

Comment:
I've gone back and forth on whether to explicitly zero out the StateMachine field here.  On the one hand, it would potentially release any references held by the state machine object earlier than they'd otherwise be released.  On the other hand, it would have a cost of zero'ing out the memory, and in most cases, the task object itself is about to get dropped, so it wouldn't actually help in most cases.  Open to other opinions.